addons/stock/res_config.py

```diff
@@ -1,24 +1,22 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from openerp.osv import fields, osv
-from openerp.tools.translate import _
+from openerp import fields, models, api, _
 
-class res_company(osv.osv):
+class ResCompany(models.Model):
     _inherit = "res.company"
-    _columns = {
-        'propagation_minimum_delta': fields.integer('Minimum Delta for Propagation of a Date Change on moves linked together'),
-        'internal_transit_location_id': fields.many2one('stock.location', 'Internal Transit Location', help="Technical field used for resupply routes between warehouses that belong to this company", on_delete="restrict"),
-    }
 
-    def create_transit_location(self, cr, uid, company_id, company_name, context=None):
+    propagation_minimum_delta = fields.Integer(string='Minimum Delta for Propagation of a Date Change on moves linked together', default=1)
+    internal_transit_location_id = fields.Many2one(comodel_name='stock.location', string='Internal Transit Location', help="Technical field used for resupply routes between warehouses that belong to this company", on_delete="restrict")
+
+    @api.model
+    def create_transit_location(self, company_id, company_name):
         '''Create a transit location with company_id being the given company_id. This is needed
            in case of resuply routes between warehouses belonging to the same company, because
            we don't want to create accounting entries at that time.
         '''
-        data_obj = self.pool.get('ir.model.data')
         try:
-            parent_loc = data_obj.get_object_reference(cr, uid, 'stock', 'stock_location_locations')[1]
+            parent_loc = self.env.ref('stock.stock_location_locations').id
```

Comment:
**parent_loc** is represent the **id** of stock locations so the name of variable should be  
```
parent_loc_id = self.env.ref('stock.stock_location_locations').id
```